### ISSUE TITLE
equality/hashcode not generated

### ISSUE THREAD

**Issue by SleepWellPupper** [PaulBraetz/RhoMicro.CodeAnalysis#35]
```cs
struct Foo;
struct Bar;
[UnionType<Foo, Bar>]
readonly partial struct FooBar;
```
This raises CS0659 and CS0661, because the `Equals(Object?)`, `==` and `!=` members are being generated but not the `Equals(FooBar)` and `GetHashCode` methods. Also, since that is the case, overload resolution for `Equals(union)` calls results in SOE when calling `Equals(Object? other)`:
```cs
public override System.Boolean Equals(System.Object? obj) =>
	obj is Success union && Equals(union); //SOE here
```
